File: R/SubstitutionsDeSaturations.R
SubstitutionsDeSaturations <- function(AS) {
  # AS <- SubstitutionsDeSaturations(AS);
  
  ngrp <- nrow(AS$GRP)
  satur <- vector("list", ngrp)
  
  for (k in seq_len(ngrp)) {
    grp <- AS$GRP[k, ]
    grp <- unlist(grp[grp != 0]) # POC check unlist
    signes <- matSignes(length(grp))
    signes <- signes[AS$declin[[k]], ]  # Ajout 18 fév 2025
    ns <- nrow(signes)
    
    sat <- vector("list", ns)  # Une liste car le nombre de lignes des matrices est le nb de variables +1
    
    for (s in seq_len(ns)) {
      sat[[s]] <- saturations(AS, 1, grp * signes[s, ])
    }
    
    satur[[k]] <- sat
  }
  
  AS$satur <- satur
  return(AS)
}

File: Matlab/SCRoF_Fevrier/SubstitutionsDeSaturations.m
function AS=SubstitutionsDeSaturations(AS)
% AS=SubstitutionsDeSaturations(AS);
ngrp=size(AS.GRP,1);
satur{ngrp}=[];
for k=1:ngrp
    grp=AS.GRP(k,:);
    grp(grp==0)=[];
    signes=matSignes(numel(grp));
    ns=size(signes,1);
    sat{ns}=[];  % une liste parce que le nombre de lignes des matrices est le nb de variables +1
    for s=1:ns
        sat{s}=saturations(AS,1,grp.*signes(s,:));
    end
    satur{k}=sat;
end
AS.satur=satur;
